sdk/core/Azure.Core.TestFramework/src/TestRecording.cs

```diff
@@ -172,7 +188,7 @@ private RecordSession Load()
 
         public void Dispose(bool save)
         {
-            if (Mode == RecordedTestMode.Record && save)
+            if (Mode == RecordedTestMode.Record && save && _session.Entries.Count > 0 && _session.Variables.Count > 0)
```

Comment:
Might be better to encapsulate this logic in `RecordSession` itself. `RecordSession.IsEmpty`?